local-macros/src/lib.rs

```diff
@@ -1465,3 +1467,115 @@ pub fn enum_unwrapped(input: TokenStream) -> TokenStream {
     }
     .into()
 }
+
+fn get_node_enum_unwrapped_call(argument: &Expr, variant_name: &Ident) -> TokenStream2 {
+    let known_node_variant_names: HashMap<String, Vec<&'static str>> =
+        HashMap::from_iter(IntoIter::new([
+            ("Expression".to_string(), vec![]),
+            ("VariableDeclarationList".to_string(), vec![]),
+            ("TypeParameterDeclaration".to_string(), vec![]),
+            ("PropertyAssignment".to_string(), vec![]),
+        ]));
+
+    let mut ancestors = vec!["Node"];
+    let variant_name_string = variant_name.to_string();
+    ancestors.extend_from_slice(known_node_variant_names.get(&variant_name_string).unwrap());
+    ancestors.push(&variant_name_string);
+    let ancestors = ancestors
+        .into_iter()
+        .map(|ancestor_str| Ident::new(ancestor_str, variant_name.span()));
+    let ancestors = quote! {
+        [#(#ancestors),*]
+    };
+
+    quote! {
+        ::local_macros::enum_unwrapped!(#argument, #ancestors)
+    }
+}
+
+struct NodeUnwrapped {
+    argument: Expr,
+    variant_name: Ident,
+}
+
+impl Parse for NodeUnwrapped {
+    fn parse(input: ParseStream) -> Result<Self> {
+        let argument: Expr = input.parse()?;
+        input.parse::<Token![,]>()?;
+        let variant_name: Ident = input.parse()?;
+        Ok(NodeUnwrapped {
+            argument,
+            variant_name,
+        })
+    }
+}
+
+#[proc_macro]
+pub fn node_unwrapped(input: TokenStream) -> TokenStream {
```

Comment:
Per comment on previous PR, I implemented `node_unwrapped!()`/`type_unwrapped!()` procedural macros on top of `enum_variant!()` that have knowledge of the `Node`/`Type` enum variants

But then I ended up feeling like the ergonomics of `.as_*()` were superior, and it seems nicer (both a bit faster to compile since it's only invoking one level of procedural macros instead of two and then the knowledge of the `Node`/`Type` variants lives outside of the procedural macro implementation) to implement those using `enum_unwrapped!()` rather than `node_unwrapped!()`/`type_unwrapped!()`